tests/test_api_hosts_delete.py

```diff
@@ -102,6 +103,89 @@ def test_create_then_delete_without_insights_id(
     assert_delete_event_is_valid(event_producer=event_producer_mock, host=host, timestamp=event_datetime_mock)
 
 
+@pytest.mark.parametrize(
+    "query_filter",
+    (
+        {"insights_id": "6e7b6317-0a2d-4552-a2f2-b7da0aece49d"},
+        {"staleness": "fresh"},
+        {"provider_type": "azure"},
+        {"provider_id": "6e7b6317-0a2d-4552-a2f2-b7da0aece49d"},
+        {"tags": "SPECIAL/tag=ToFind"},
+        {"display_name": "hbi_display.redhat.com"},
+        {"tags": "ns1/key2=val1", "display_name": "hbi_display.redhat.com"},
+        {"hostname_or_id": "test.server.redhat.com"},
+        {"fqdn": "test.server.redhat.com"},
+    ),
+)
+def test_delete_hosts_using_filter(
+    event_producer_mock,
+    db_create_multiple_hosts,
+    db_get_hosts,
+    api_delete_filtered_hosts,
+    query_filter,
+    host_ids_xjoin_post,
+):
+
+    extraData = {
+        "canonical_facts": {
+            "insights_id": "6e7b6317-0a2d-4552-a2f2-b7da0aece49d",
+            "provider_id": "6e7b6317-0a2d-4552-a2f2-b7da0aece49d",
+            "provider_type": "azure",
+            "fqdn": "test.server.redhat.com",
+        },
+        "tags": {"ns1": {"key1": ["val1", "val2"], "key2": ["val1"]}, "SPECIAL": {"tag": ["ToFind"]}},
+        "display_name": "hbi_display.redhat.com",
+    }
+
+    created_hosts = db_create_multiple_hosts(
+        how_many=len(XJOIN_HOSTS_RESPONSE_FOR_FILTERING["hosts"]["data"]), extra_data=extraData
+    )
+    host_ids = [str(host.id) for host in created_hosts]
```

Comment:
You could cast `XJOIN_HOSTS_RESPONSE_FOR_FILTERING["hosts"]["data"]` to a list of hosts and call `db_create_multiple_hosts()` using the `hosts` keyword argument to avoid the extra data and id switching.
```suggestion
    hosts_to_create = [Host(**host_init_data) for host_init_data in XJOIN_HOSTS_RESPONSE_FOR_FILTERING["hosts"]["data"]]
    created_hosts = db_create_multiple_hosts(hosts=hosts_to_create)
```